kf_task_fhir_etl/common/utils.py

```diff
@@ -0,0 +1,91 @@
+import os
+
+from dotenv import find_dotenv, load_dotenv
+from d3b_utils.requests_retry import Session
+from requests import RequestException
+
+DOTENV_PATH = find_dotenv()
+if DOTENV_PATH:
+    load_dotenv(DOTENV_PATH)
+
+
+FHIR_COOKIE = os.getenv("FHIR_COOKIE")
+FHIR_USERNAME = os.getenv("FHIR_USERNAME")
+FHIR_PASSWORD = os.getenv("FHIR_PASSWORD")
+
+
+def not_none(val):
+    if val is None:
+        raise ValueError("Missing required value")
+    return val
+
+
+def drop_none(body):
+    return {k: v for k, v in body.items() if v is not None}
+
+
+def yield_resources(host, endpoint, filters, show_progress=False):
+    """Scrapes the FHIR service for paginated entities matching the filter params.
+    Note: It's almost always going to be safer to use this than requests.get
+    with search parameters, because you never know when you'll get back more
+    than one page of results for a query.
+
+    :param host: A FHIR service base URL (e.g. "http://localhost:8000")
+    :type host: str
+    :param endpoint: A FHIR service endpoint (e.g. "Patient")
+    :type endpoint: str
+    :param filters: dict of filters to winnow results from the FHIR service
+        (e.g. {"name": "Children\'s Hospital of Philadelphia"})
+    :type filters: dict
+    :raises Exception: If the FHIR service doesn't return status 200
+    :yields: resources matching the filters
+    """
+    url = f"{host.rstrip('/')}/{endpoint.lstrip('/')}"
+
+    expected = 0
+    link_next = url
+    found_resource_ids = set()
+
+    headers = {"Content-Type": "application/fhir+json;charset=utf-8"}
+    auth = None
+
+    if FHIR_COOKIE:
+        headers["Cookie"] = FHIR_COOKIE
+
+    if FHIR_USERNAME and FHIR_PASSWORD:
+        auth = (FHIR_USERNAME, FHIR_PASSWORD)
+
+    while link_next is not None:
+        resp = Session().get(link_next, params=filters, headers=headers, auth=auth)
+
```

Comment:
Maybe instantiate Session before while loop
```suggestion
    session = Session()
    while link_next is not None:
        resp = session.get(link_next, params=filters, headers=headers, auth=auth)
```

Reply:
I made this change, thanks!